gsoc_final_post/inigo_tellaetxe_gsoc_2024_final_report.rst

```diff
@@ -0,0 +1,226 @@
+      :align: center
+      :width: 800
+  
+    We see that the Keras result is very similar to the PyTorch result, though the fiber crossings are not as well resolved in the Keras result. This is not a thing to worry since this is not systematic, as seen in other sets of unseen data.
+
+
+* **Implemented a Variational AutoEncoder (VAE) architecture based on the** `FINTA <https://doi.org/10.1016/j.media.2021.102126>`_ **AE architecture.**
+
+  * After thorough discussion with my mentors, we decided that regularizing the latent space could be a good idea due to two reasons: to avoid "white spaces" in the latent space it that could lead undefined points due to the possible discontinuity of the space; and to prevent the model from learning the identity function, what is a known issue with unregularized AutoEncoders, though it does not always happen. The only addition to the original AE architecture was the inclusion of the Kullback-Leibler divergence loss term in the loss function, and two dense layers after the convolutional layers of the encoder, to output the ::math:`\mu` and ::math:`\sigma^2` of the latent space vector ::math:`z`.
+
+  * The model is found in the ``vae_model.py`` module of the `TractoEncoder GSoC <https://github.com/itellaetxe/tractoencoder_gsoc>`_ repository. The architecture can be summarized in the figure below:
+
+  .. image:: https://github.com/dipy/dipy.org/blob/master/_static/images/gsoc/2024/inigo/inigo_variational_autoencoder.png
+    :alt: VAE architecture diagram.
+    :align: center
+    :width: 800
+
+  * It was necessary to implement Batch Normalization after the Encoder convolutional layers and exponential operator clipping, to prevent gradient explosion while training the model. The figure above displays that the encoder outputs the variance of the latent space vector (::math:`\sigma^2`), though it is a common practice to output the log-variance instead. This leads to using ::math:`\sigma = e^{\frac{1}{2}\text{log}{\sigma^2}}`, which can get numerically quite unstable as it can shoot up very quickly when a sufficiently big log-variance is input. This modification allowed stable training and it was a major contribution to the robustness of the architecture. Shoutout to my lab colleague `Jorge <https://github.com/jgarciacondado>`_ for the ideas and discussions on the topic.
+  * Weighing the Kullback-Leibler loss component was also implemented, based on the `Beta-VAE <https://openreview.net/forum?id=Sy2fzU9gl>`_ work, aiming for a stronger disentanglement of the latent space. However, this parameter was never explored (it was always set to 1.0) due to its trade-off with the reconstruction accuracy, but it is a potential improvement for future work in the context of hyperparameter optimization.
+  * The figure below shows a summary of the VAE results, also using the FiberCup dataset:
+
+    .. image:: https://github.com/dipy/dipy.org/blob/master/_static/images/gsoc/2024/inigo/vanilla_vae_120_epoch_results.png
```

Comment:
It could be better to say 'Training Data' instead of 'Original Data' (since the next figure says reconstruction of training data). Also, instead of showing #1 and #2 reconstruction, it might be better to show the original #1 and the reconstructed #1. Hopefully it is not too late to edit the figure?

Reply:
I agree, thanks for the suggestion. I will edit the figure because I saved it as a draw.io file, so I can easily edit it :)